astroplan/constraints.py

```diff
@@ -394,7 +395,81 @@ def compute_constraint(self, times, observer, targets):
                              "MoonSeparationConstraint.")
         return mask
 
+    
+class DarkGreyBrightConstraint(Constraint):
+    """
+    Constrain whether it is dark, gray or bright time or how bright.
+    """
+    # TODO: make a new test
+    @u.quantity_input(horizon=u.deg)
+    def __init__(self, lunar_horizon=90*u.deg, min_moon_illumination=None,
+                                                max_moon_illumination=None):
+        '''
+        Parameters
+        ----------
+        Lunar_horizon : `~astropy.units.Quantity` (optional)
+            The altitude of the ``horizon`` and defaults to 90 degrees       
```

Comment:
The only issue with this is that most astronomers have a different conception of what dark, grey and bright mean. Dark time is moon illuminations < 0.25, even if the moon is up.

And then, since this is always going to be used as a constraint on the brightness of the night sky, it doesn't make sense to assume the moon is up, since the night sky can be darker when the Moon is 65% illuminated, but below the horizon, than it is when the moon is up but 30% illuminated.